source/js/components/join/join.jsx

```diff
@@ -224,28 +224,48 @@ export default class JoinUs extends React.Component {
    * Render the email field in signup CTA.
    */
   renderEmailField() {
-    let classes = classNames(`mb-2`, {
+    let wrapperClasses = classNames(``, {
       "has-danger":
         (!this.state.apiSuccess &&
           this.state.userTriedSubmitting &&
           !this.validatesAsEmail(this.email.value)) ||
         this.state.signupFailed
     });
 
+    let classes = classNames(``, {
+      "position-relative": wrapperClasses != ``
+    });
+
     return (
-      <div className={classes}>
-        <input
-          type="email"
-          className="form-control"
-          placeholder="Enter email address"
-          ref={el => (this.email = el)}
-          onFocus={evt => this.onInputFocus(evt)}
-        />
+      <div className={wrapperClasses}>
+        <div className={classes}>
+          <input
+            type="email"
+            className="form-control align-items"
+            placeholder="Enter email address"
+            ref={el => (this.email = el)}
+            onFocus={evt => this.onInputFocus(evt)}
+          />
+          {this.state.userTriedSubmitting &&
+            !this.state.apiSubmitted &&
+            !this.validatesAsEmail(this.email.value) && (
+              <div className="glyph-container">
+                <span className="form-error-glyph" />
+              </div>
+            )}
+        </div>
+        {this.state.userTriedSubmitting &&
+          (this.email.value == "" || this.email.value == null) && (
+            <p className="body-small form-check form-control-feedback">
+              This is a required section.
+            </p>
+          )}
         {this.state.userTriedSubmitting &&
           !this.state.apiSubmitted &&
-          !this.validatesAsEmail(this.email.value) && (
+          !this.validatesAsEmail(this.email.value) &&
+          (!this.email.value == "" || !this.email.value == null) && (
             <p className="body-small form-check form-control-feedback">
-              Please enter your email
+              Please enter a valid email address.
```

Comment:
This block is quite confusing (Line 257-270)


```jsx

{this.state.userTriedSubmitting &&
  (this.email.value == "" || this.email.value == null) && (
    <p className="body-small form-check form-control-feedback">
      This is a required section.
    </p>
  )}
{this.state.userTriedSubmitting &&
  !this.state.apiSubmitted &&
  !this.validatesAsEmail(this.email.value) &&
  (!this.email.value == "" || !this.email.value == null) && (
    <p className="body-small form-check form-control-feedback">
      Please enter a valid email address.
    </p>
  )}
```

Why are we checking `this.state.apiSubmitted` just for the one case and not the other? (You might want to verify if checking `this.state.apiSubmitted` is really necessary.)

Since the two error messages is mutually exclusive we should be able to merge & simplify the above code into something like

```jsx
{this.state.userTriedSubmitting && (
  <p className="body-small form-check form-control-feedback">
    {figure out what error message to show here}
  </p>
)}
```

Reply:
The first error message is for when the input is empty vs the second error for the when there is text in the input but it's not a valid email address.